### PR TITLE
chore(types): convert a few files to typescript

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
@kinwa91 this is one of the repos that needs some kokoro love :)

File: dev/conformance/runner.ts [hunk 1]
```diff
@@ -39,32 +39,34 @@ const gax = require('google-gax');
 const grpc = new gax.GrpcClient().grpc;
 
 /** List of test cases that are ignored. */
-const ignoredRe = [];
+// tslint:disable-next-line:no-any
+const ignoredRe: any[] = [];
```
Comment (schmidt-sebastian) [inline dev/conformance/runner.ts]: This can be `RegExp[]`.

File: dev/src/convert.ts [hunk 2]
```diff
@@ -225,9 +226,13 @@ function convertDocument(document) {
...
-  timestampFromJson: convertTimestamp,
-  valueFromJson: convertValue,
-  detectValueType: detectValueType,
-};
+// module.exports = {
+//   documentFromJson: convertDocument,
+//   timestampFromJson: convertTimestamp,
+//   valueFromJson: convertValue,
+//   detectValueType: detectValueType,
+// };
```
Comment (schmidt-sebastian) [inline dev/src/convert.ts]: We can probably just remove this.

File: dev/src/convert.ts [hunk 3]
```diff
@@ -44,19 +44,20 @@ const validate = createValidator();
  * @return {{nanos,seconds}|undefined} The value as expected by Protobuf JS or
  * undefined if no input was provided.
  */
-function convertTimestamp(timestampValue, argumentName) {
-  let timestampProto = undefined;
+function convertTimestamp(timestampValue, argumentName?) {
+  // tslint:disable-next-line:no-any
+  let timestampProto: any = undefined;
```
Comment (schmidt-sebastian) [inline dev/src/convert.ts]: You should be able to import the type definitions for this from https://github.com/googleapis/nodejs-firestore/blob/master/dev/protos/firestore_proto_api.d.ts

File: dev/conformance/runner.ts [hunk 4]
```diff
@@ -384,12 +378,11 @@ function runTest(spec) {
...
 
-  const setTest = function(spec) {
+  const setTest = spec => {
     const overrides = {commit: commitHandler(spec)};
-
     return createInstance(overrides).then(() => {
-      const setOption = {};
-
+      // tslint:disable-next-line:no-any
+      const setOption: any = {};
```
Comment (schmidt-sebastian) [inline dev/conformance/runner.ts]: Can you make this `AnyDuringMigration`? I should be able to clean this up later.

File: dev/conformance/runner.ts [hunk 5]
```diff
@@ -248,7 +250,8 @@ function commitHandler(spec) {
     try {
       assert.deepEqual(request, convertProto.commitRequest(spec.request));
 
-      const res = {
+      // tslint:disable-next-line:no-any
+      const res: any = {
```
Comment (schmidt-sebastian) [inline dev/conformance/runner.ts]: This is https://github.com/googleapis/nodejs-firestore/blob/master/dev/protos/firestore_proto_api.d.ts#L2035

File: dev/conformance/runner.ts [hunk 6]
```diff
@@ -39,32 +39,34 @@ const gax = require('google-gax');
...
 
 /** List of test cases that are ignored. */
-const ignoredRe = [];
+// tslint:disable-next-line:no-any
+const ignoredRe: any[] = [];
 
 /** If non-empty, list the test cases to run exclusively. */
-const exclusiveRe = [];
+// tslint:disable-next-line:no-any
+const exclusiveRe: any[] = [];
```
Comment (schmidt-sebastian) [inline dev/conformance/runner.ts]: Same here.

**Comment by JustinBeckwith** [comment]
thank you for the awesome PR feedback :)

File: dev/src/convert.ts [hunk 7]
```diff
@@ -67,14 +67,14 @@ function convertTimestamp(timestampValue, argumentName?) {
...
+      seconds,
+      nanos,
     };
   } else if (is.defined(timestampValue)) {
     validate.isObject('timestampValue', timestampValue);
     timestampProto = {
-      seconds: timestampValue.seconds || undefined,
-      nanos: timestampValue.nanos || undefined,
+      seconds: timestampValue.seconds,
+      nanos: timestampValue.nanos,
```
Comment (schmidt-sebastian) [inline dev/src/convert.ts]: I think that this change my break the conformance tests (either now or in the future), which require the absolute minimum proto. ` timestamp { seconds: 0, nanos: 0 }` might fail tests that require `timestamp {} ` (in JSON format, not Proto3).

This might be totally wrong, but it's the only explanation I could come up with why I might have written this code this way.

File: dev/src/convert.ts [hunk 8]
```diff
@@ -66,8 +67,8 @@ function convertTimestamp(timestampValue, argumentName) {
     }
 
     timestampProto = {
-      seconds: seconds || undefined,
-      nanos: nanos || undefined,
+      seconds,
```
Comment (schmidt-sebastian) [inline dev/src/convert.ts]: Can you also revert the change here that removes "undefined"? Thanks.
